### PR TITLE
Wip deprecate auto fusing setting johanandren

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22431

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala [hunk 1]
```diff
@@ -347,6 +347,10 @@ case class PhasedFusingActorMaterializer(
     _logger.warning("Fuzzing mode is enabled on this system. If you see this warning on your production system then " +
       "set akka.stream.materializer.debug.fuzzing-mode to off.")
   }
+  if (!settings.autoFusing) {
+    _logger.info("Deprecated setting auto-fusing set to false. Since Akka 2.5.0 it does not have any effect " +
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala]: I think we have used `warning` level for such log messages in other places

**Review by patriknw — CHANGES_REQUESTED** [review]

File: akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala (hunk 1 above)
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala]: +1 for warning level, LGTM otherwise

**Review by ktoso — APPROVED** [review]

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 2]
```diff
@@ -306,6 +306,15 @@ In 2.4 an explicit `async` marker (``AsyncBoundary`` attribute) had to be added.
 defined ``blocking-io-dispatcher`` as default followed by a ``map`` will now be separated by an async boundary, which was not
 the case in 2.4.
 
+Removal of the auto-fuse setting
+--------------------------------
+
+In 2.4 fusing stages together into the same actor could be completely disabled with the setting
+``akka.stream.materializer.auto-fusing``. The new materializer introduced in Akka 2.5 does not support disabling fusing,
+so this setting does not have any effect any more and has been deprecated. Running each stage in a stream can be done
```
Comment (2m) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: Missing a note about running stages on separate actors.

> ... running each statge in a stream *on a separate actor* can be done ...

**Review by 2m — CHANGES_REQUESTED** [review]
LGTM modulus small doc improvement.

**Review by 2m — APPROVED** [review]

**Pull request merged**